### DISCUSSION TITLE
Regex manager and docker image parsing

### DISCUSSION THREAD

**Discussion by lukasmrtvy** [renovatebot/renovate#12434] [Request Help]
Hey 👋 How to parse docker image, which can be in format `mycustomregistry.com/foo/bar:latest@sha256:a4cea2f286c9ad5a5db83ae932504dfc6e1de359833b9d4fc9f0eba80d08aa23`, I guess that theres no need to write regexp, cuz thats widely used format in renovate, right? Can config like this work? Thanks

**renovate.json**
```
  "regexManagers": [
    {
      "fileMatch": [".*.nomad"],
      "matchStrings": ["image = \"(.*)\""],
      "depTypeTemplate": "image",
      "datasourceTemplate": "docker"
    }
  ]
```

**foo.nomad**
```
job "foo" {
  group "cache" {
    task "redis" {
      driver = "docker"
      config {
        image = "redis:5.0.12"
        port_map {
          db = 6379
        }
      }
    }
  }
}
```

**Comment by rarkins** [comment]
The format of that string looks standard, yes. Your regex doesn't look right, so I suggest you experiment in a test repo until you get it how you like it. In particular use named regex capture groups

**Reply by lukasmrtvy** [reply]
Renovate does not have regular expression to parse standard docker image format ? Even with (optional) groups with `.*` matching   ?

**Reply by rarkins** [reply]
Look at some of the examples in https://docs.renovatebot.com/configuration-options/#regexmanagers, particularly how `(?<currentValue>.*)` is used
